### DISCUSSION TITLE
I'm having trouble with binding.

### DISCUSSION THREAD

**Discussion by IronEaterBeast** [AvaloniaUI/Avalonia#15412] [Q&A]
1. My window is bound using the following binding: DataContext = MainWindowViewModel;
2. I have a DataGrid: ItemsSource="{Binding DomainOC}"
3. The ComboBox has a column that is a ComboBox.
4. The ComboBox's SelectedItem is MainWindowViewModel.DomainOC.CountrySelected, and its ItemsSource is MainWindowViewModel.CountryOC, which is a parent level property of DomainOC.

In other words, I need to access the parent MainWindowViewModel's CountryOC property from inside the DataGrid to give all rows the same set of ComboBox options. I've tried a lot of code, such as:

`<ComboBox ItemsSource="{Binding RelativeSource={RelativeSource Mode=FindAncestor,  AncestorType={x:Type  vm:MainWindowViewModel}}, Path=CountryOC}" SelectedItem="{Binding CountrySelected}">`
`<ComboBox ItemsSource="{Binding RelativeSource={RelativeSource FindAncestor,AncestorType={x:Type Window}},Path=DataContext.CountryOC}" SelectedItem="{Binding CountrySelected}">`

But none of them work. Any suggestions?

**Comment by rabbitism** [comment]
https://docs.avaloniaui.net/docs/basics/data/data-binding/compiled-bindings#type-casting
